blog/_src/posts/2019-08-14-lexical-and-dynamic-scope.md

```diff
@@ -0,0 +1,247 @@
+    functions.[^6] He considers both the "downward" case (where a function is
+    passed to another function) and the "upward" case (where a function returns
+    a function), and remarks on the correspondence between Lisp's FUNARG device
+    and Landin's closures.
+
+  * **1975**. Gerald Sussman and Guy Steele publish the first Scheme paper.[^7]
+    They describe their goal of a Lisp-like language that is based on the
+    lambda calculus. As a consequence, they implement lexical scoping with
+    closures, to preserve the substitution semantics of the lambda calculus.
+    They compare this scoping discipline to ALGOL's.
+
+  * **1978**. Steele and Sussman describe various programming language design
+    choices, by developing an interpreter for each programming language
+    variation.[^8] In particular, they provide a detailed discussion on
+    lexical and dynamic scoping.
+
+[^2]: J. McCarthy. "Recursive Functions of Symbolic Expressions and Their
+Computation by Machine, Part I," _Communications of the ACM_, vol. 3, no. 4,
+April 1960. \[[DOI](https://doi.org/10.1145/367177.367199)\]\[[Available
+online](http://jmc.stanford.edu/articles/recursive/recursive.pdf)\]
+
+[^3]: J. McCarthy. "History of LISP," in _History of Programming Languages_,
+1978. \[[DOI](https://doi.org/10.1145/800025.1198360 )\]\[[Available
+online](http://jmc.stanford.edu/articles/lisp/lisp.pdf)\]
+
+[^4]: P. Naur (ed.). "Revised Report on Algorithmic Language ALGOL 60,"
+_Communications of the ACM_, vol. 6, no. 1, January 1963.
+\[[DOI](http://dx.doi.org/10.1145/366193.366201)\]\[[Available
+online](https://www.masswerk.at/algol60/report.htm)\]
+
+[^5]: P. Landin. "The mechanical evaluation of expressions," _The Computer
+Journal_, vol. 6, no. 4, January 1964.
+\[[DOI](https://doi.org/10.1093/comjnl/6.4.308)\]\[[Available
+online](https://www.cs.cmu.edu/~crary/819-f09/Landin64.pdf)\]
+
+[^6]: J. Moses. "The Function of FUNCTION in LISP or Why the FUNARG Problem
+Should be Called the Environment Problem," _SIGSAM Bulletin 15_, July 1970.
+\[[DOI](https://doi.org/10.1145/1093410.1093411)\]\[[Available
+online](https://dspace.mit.edu/handle/1721.1/5854)\]
+
+[^7]: G. Sussman and G. Steele. "SCHEME: An Interpreter for Extended Lambda
+Calculus." 1975. \[[Available
+online](https://dspace.mit.edu/handle/1721.1/5794)\]
+
+[^8]: G. Steele and G. Sussman. "The Art of the Interpreter or, The Modularity
+Complex (Parts Zero, One, and Two)." 1978. \[[Available
+online](https://dspace.mit.edu/handle/1721.1/6094)\]
+
+
+## Next stop, R
+
+Now that we have examined the definitions of lexical and dynamic scope, and
+also explored some history, we are ready to return to the original question.
+_Is R lexically or dynamically scoped?_
+
+In the next blog post, we'll answer that question, and also see how R can be
+very confusing.
+
+_I would like to thank Ben Greenman and Artem Pelenitsyn for their feedback on
+an earlier draft of this post._
+
+---
```

Comment:
Delete this line and the heading below it?

Reply:
I'll keep the line and remove the heading. It looks weird in Markdown, but I think the line is necessary when rendered:

Line+heading:
![image](https://user-images.githubusercontent.com/547059/63362075-3f11d500-c33f-11e9-90f0-3d72129a0669.png)

Line only:
![image](https://user-images.githubusercontent.com/547059/63362228-80a28000-c33f-11e9-97de-6fc4b8f784ce.png)

No line, no heading:
![image](https://user-images.githubusercontent.com/547059/63362155-62d51b00-c33f-11e9-925b-2a08a1231bcd.png)